insights/tests/client/connection/test_checkin.py

```diff
@@ -191,3 +191,14 @@ def test_response_failure(get_proxies, post, get_canonical_facts, rm_conf, statu
 
     with raises(Exception):
         connection.checkin()
+
+
+def test_unregistered_host():
```

Comment:
Let’s patch the function that checks whether the host is registered.

```suggestion
@patch("insights.client.connection.InsightsConnection.api_registration_check", return_value=false)
def test_unregistered_host(api_registration_check):
```

or 

```suggestion
@patch("insights.client.connection.InsightsConnection._fetch_system_by_machine_id", return_value=false)
def test_unregistered_host(fetch_system_by_machine_id):
```

if you accepted my suggestion above.

All tests in the file patch `_init_session`, because it verifies RHSM certificates on the host machine. That can cause false errors.

```suggestion
@patch("insights.client.connection.InsightsConnection._init_session", Mock())
def test_unregistered_host():
```

They patch `InsightsUploadConf.get_rm_conf` too. It seems not required: it may be a mistake or a relic. Let’s go the safe way and keep the patches consistent. They can be removed in bulk separately.

```suggestion
@patch('insights.client.connection.InsightsUploadConf.get_rm_conf', return_value={})
def test_unregistered_host(rm_conf):
```

Combined:

```suggestion
@patch('insights.client.connection.InsightsUploadConf.get_rm_conf', return_value={})
@patch("insights.client.connection.InsightsConnection._init_session", Mock())
@patch("insights.client.connection.InsightsConnection._fetch_system_by_machine_id", return_value=false)
test_unregistered_host(fetch_system_by_machine_id, rm_conf):
```

---

The other tests need to be updated too. A patch to api_registration_check` / `_fetch_system_by_machine_id` with a truth return value needs to be added for several reasons:

* The non-patched original method would issue real requests and tamper with the machine ID on the host.
* The tests need to ensure that the check passes, so they can check what happens after the conditional.
* If there was a test with a patched `get` method, assertions for its calls would break.

```python
...
@patch("insights.client.connection.InsightsConnection._fetch_system_by_machine_id", return_value=True)
...
def test_canonical_facts_request(get_proxies, post, fetch_system_by_machine_id, get_canonical_facts, rm_conf):
```